testing/internal/e2e/host/aws/dynamichostcatalog_test.go

```diff
@@ -0,0 +1,413 @@
+package aws_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/hashicorp/boundary/api/hostcatalogs"
+	"github.com/hashicorp/boundary/api/hosts"
+	"github.com/hashicorp/boundary/api/hostsets"
+	"github.com/hashicorp/boundary/api/targets"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type config struct {
+	AwsAccessKeyId     string `envconfig:"E2E_AWS_ACCESS_KEY_ID"`
+	AwsSecretAccessKey string `envconfig:"E2E_AWS_SECRET_ACCESS_KEY"`
+	AwsHostSetFilter1  string `envconfig:"E2E_AWS_HOST_SET_FILTER1"`
+	AwsHostSetCount1   string `envconfig:"E2E_AWS_HOST_SET_COUNT1"`
+	AwsHostSetIps1     string `envconfig:"E2E_AWS_HOST_SET_IPS1"`
+	AwsHostSetFilter2  string `envconfig:"E2E_AWS_HOST_SET_FILTER2"`
+	AwsHostSetCount2   string `envconfig:"E2E_AWS_HOST_SET_COUNT2"`
+	TargetSshKeyPath   string `envconfig:"E2E_SSH_KEY_PATH"` // e.g. /Users/username/key.pem
+	TargetSshUser      string `envconfig:"E2E_SSH_USER"`     // e.g. ubuntu
+	TargetPort         string `envconfig:"E2E_SSH_PORT" default:"22"`
+}
+
+func (c *config) validate() error {
+	if c.AwsAccessKeyId == "" {
+		return errors.New("AwsAccessKeyId is empty. Set environment variable: E2E_AWS_ACCESS_KEY_ID")
+	}
```

Comment:
FYI you can set `required:"true"` in your struct tags if you want this enforced by `envconfig`: https://github.com/kelseyhightower/envconfig#struct-tag-support.

Reply:
Oh. Perfect.